product/web3modal/src/main/kotlin/com/walletconnect/web3/modal/ui/components/button/Web3ModalState.kt

```diff
@@ -41,12 +39,20 @@ class Web3ModalState(
     private val coroutineScope: CoroutineScope,
     private val navController: NavController
 ) {
+    private val logger: Logger = wcKoinApp.koin.get()
     private val observeSelectedChainUseCase: ObserveSelectedChainUseCase = wcKoinApp.koin.get()
     private val observeSessionTopicUseCase: ObserveSessionTopicUseCase = wcKoinApp.koin.get()
     private val getSessionTopicUseCase: GetSessionTopicUseCase = wcKoinApp.koin.get()
     private val getSelectedChainUseCase: GetSelectedChainUseCase = wcKoinApp.koin.get()
 
-    val isOpen = navController.currentBackStackEntryFlow.mapLatest { it.destination.route?.startsWith(Route.WEB3MODAL.path) ?: false }
+    val isOpen = ComponentDelegate.modalComponentEvent
+        .map { event ->
+            when (event) {
+                ComponentEvent.ModalHiddenEvent -> false
+                ComponentEvent.ModalExpandedEvent -> true
+            }
+        }
```

Comment:
If you want to use the suggestion above, then this becomes

```suggestion
    val isOpen = ComponentDelegate.modalComponentEvent
        .map { event -> event.isOpen }
```